src/PowerShellEditorServices.Engine/Services/TextDocument/CompletionResults.cs

```diff
@@ -0,0 +1,340 @@
+            {
+                CompletionText = completionText,
+                CompletionType = completionType,
+                ListItemText = listItemText,
+                ToolTipText = toolTipText,
+                SymbolTypeName = symbolTypeName
+            };
+        }
+
+        #endregion
+
+        #region Public Methods
+
+        /// <summary>
+        /// Compares two CompletionResults instances for equality.
+        /// </summary>
+        /// <param name="obj">The potential CompletionResults instance to compare.</param>
+        /// <returns>True if the CompletionResults instances have the same details.</returns>
+        public override bool Equals(object obj)
+        {
+            CompletionDetails otherDetails = obj as CompletionDetails;
+            if (otherDetails == null)
+            {
+                return false;
+            }
+
+            return
+                string.Equals(this.CompletionText, otherDetails.CompletionText) &&
+                this.CompletionType == otherDetails.CompletionType &&
+                string.Equals(this.ToolTipText, otherDetails.ToolTipText) &&
+                string.Equals(this.SymbolTypeName, otherDetails.SymbolTypeName);
+        }
+
+        /// <summary>
+        /// Returns the hash code for this CompletionResults instance.
+        /// </summary>
+        /// <returns>The hash code for this CompletionResults instance.</returns>
+        public override int GetHashCode()
+        {
+            return
+                string.Format(
+                    "{0}{1}{2}{3}{4}",
+                    this.CompletionText,
+                    this.CompletionType,
+                    this.ListItemText,
+                    this.ToolTipText,
+                    this.SymbolTypeName).GetHashCode();
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        private static CompletionType ConvertCompletionResultType(
+            CompletionResultType completionResultType)
+        {
+            switch (completionResultType)
+            {
+                case CompletionResultType.Command:
+                    return CompletionType.Command;
+
+                case CompletionResultType.Method:
+                    return CompletionType.Method;
+
+                case CompletionResultType.ParameterName:
+                    return CompletionType.ParameterName;
+
+                case CompletionResultType.ParameterValue:
+                    return CompletionType.ParameterValue;
+
+                case CompletionResultType.Property:
+                    return CompletionType.Property;
+
+                case CompletionResultType.Variable:
+                    return CompletionType.Variable;
+
+                case CompletionResultType.Namespace:
+                    return CompletionType.Namespace;
+
+                case CompletionResultType.Type:
+                    return CompletionType.Type;
+
+                case CompletionResultType.Keyword:
```

Comment:
I wonder if we should map `DynamicKeyword` to `CompletionType.Keyword`?

Reply:
Do you think those are falling into `CompletionType.Unknown`?

Reply:
I would suppose so but to be honest this is a feature (dynamic keywords) where I have little experience.  Just seemed like low hanging fruit. :-)

Reply:
I wonder if that could be causing some issues with loading of completions of dynamic parameters... probably not. I'll still add it. Thanks!

Reply:
added!

Reply:
Probably not causing dynamic parameter issues. Unless you're using DSC, it's unlikely you'd see any issue here